executor/src/witgen/machines/fixed_lookup_machine.rs

```diff
@@ -24,154 +23,135 @@ use crate::Identity;
 
 use super::{Connection, ConnectionKind, Machine};
 
-type Application = (Vec<PolyID>, Vec<PolyID>);
-type Index<T> = BTreeMap<Vec<T>, IndexValue>;
+/// An Application specifies a lookup cache.
+#[derive(Hash, Eq, PartialEq, Ord, PartialOrd, Clone)]
+struct Application {
+    pub identity_id: u64,
+    /// Booleans indicating if the respective column is a known input column (true)
+    /// or an unknown output column (false).
+    pub inputs: Vec<bool>,
```

Comment:
not necessary to change here, but I wonder if this would be faster with a bitvector

Reply:
I can check. I think bitvector is mostly compact, but slow.

Reply:
yea fine with keeping